Consider uploading source dist (aka. "sdist") to PyPI

**Is your feature request related to a problem? Please describe.**
<!-- A clear and concise description of what the problem is. E.g., I'm always frustrated when [...] -->

I'm trying to create a conda package for this utility.  The recommended way to get this done is via [grayskull](https://github.com/conda-incubator/grayskull).  For that tool to correctly operate, it requires the sdist package to be uploaded to PyPI.

**Describe the solution you'd like**
<!-- A clear and concise description of what you want to happen. -->

Include the "sdist" source package when you do PyPI uploads of new releases.

**Describe alternatives you've considered**
<!-- A clear and concise description of any alternative solutions or features you've considered, if any. -->

Currently, I got away with conda-packaging using GitHub's archive for extracting the tarball of the latest release.  It works, but a cleaner cut would be to have the sdist regularly uploaded to PyPI.

**Additional context**
<!--
  Add any other context or screenshots about the feature request here. If there's a
  specific inventory file that helps demonstrate the behavior you want, attach it here.
-->

Before I test the PyPI sdists, I'd like to get a functional build, including testing.  However, I'm stuck getting this package to test properly (in a condo-build environment):  I have a few pytest markers missing (`pytest.mark.api` and `pytest.mark.cli`).  Where are these coming from?

```
+ pytest -rsxX -Werror
============================= test session starts ==============================
platform darwin -- Python 3.11.0, pytest-7.2.0, pluggy-1.0.0
rootdir: $SRC_DIR
plugins: timeout-2.1.0, ordering-0.6
collected 291 items / 8 errors

==================================== ERRORS ====================================
___________________ ERROR collecting tests/test_api_fail.py ____________________
tests/test_api_fail.py:41: in <module>
    pytestmark = [pytest.mark.api, pytest.mark.local]
../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/lib/python3.11/site-packages/_pytest/mark/structures.py:544: in __getattr__
    warnings.warn(
E   pytest.PytestUnknownMarkWarning: Unknown pytest.mark.api - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
```

Any ideas?

Thanks for all the tips.  I think I have an almost OK package.  I still face this:

```text
_ TestInventory.test_api_inventory_bytes_fname_instantiation[no_op--plaintext] _

self = <tests.test_api_good.TestInventory object at 0x102813810>
source_type = <SourceTypes.BytesPlaintext: 'bytes_plain'>, inv_arg = 'plaintext'
path_fxn = <function no_op at 0x1024e68e0>
res_path = PosixPath('tests/resource')
misc_info = <conftest.misc_info.<locals>.Info object at 0x102786890>
attrs_inventory_test = <function attrs_inventory_test.<locals>.func at 0x1029980e0>
check = <pytest_check.context_manager.CheckContextManager object at 0x101a64850>

    @pytest.mark.parametrize(
        ["source_type", "inv_arg"],
        [
            (soi.SourceTypes.BytesPlaintext, "plaintext"),
            (soi.SourceTypes.BytesZlib, "zlib"),
            (soi.SourceTypes.FnamePlaintext, "fname_plain"),
            (soi.SourceTypes.FnameZlib, "fname_zlib"),
        ],
        ids=(lambda v: v if isinstance(v, str) else ""),
    )
    @pytest.mark.parametrize("path_fxn", PATH_FXNS, ids=PATH_FXN_IDS)
    def test_api_inventory_bytes_fname_instantiation(
        self,
        source_type,
        inv_arg,
        path_fxn,
        res_path,
        misc_info,
        attrs_inventory_test,
        check,
    ):
        """Check bytes and filename modes for Inventory instantiation."""
        fname = misc_info.FNames.RES

        if source_type in (
            soi.SourceTypes.BytesPlaintext,
            soi.SourceTypes.FnamePlaintext,
        ):
            fname += misc_info.Extensions.DEC
        else:
            fname += misc_info.Extensions.CMP

        source = path_fxn(res_path / fname)

        if source_type in (soi.SourceTypes.BytesPlaintext, soi.SourceTypes.BytesZlib):
            # Passing in the actual inventory contents, and not just the location
            source = soi.readbytes(source)

        # General import, without a specified kwarg
>       with check.check(msg="general"):
E       AttributeError: 'CheckContextManager' object has no attribute 'check'
```

Any ideas on how to fix it?

Brian, with the package you provided, and pytest-check==1.1.3, I get 12 errors that look like this:

```text
_____________________ TestCore.test_api_decompress[no_op] ______________________

self = <tests.test_api_good.TestCore object at 0x1030630d0>
path_fxn = <function no_op at 0x102d16840>
scratch_path = PosixPath('/private/var/folders/y4/n6m6rkls6zs72x8jd2w2tlx40000gn/T/pytest-of-andre/pytest-13/test_api_decompress_no_op_0')
misc_info = <conftest.misc_info.<locals>.Info object at 0x102e42250>
decomp_cmp_test = <function decomp_cmp_test.<locals>.func at 0x102fb77e0>

    @pytest.mark.parametrize("path_fxn", PATH_FXNS, ids=PATH_FXN_IDS)
    def test_api_decompress(self, path_fxn, scratch_path, misc_info, decomp_cmp_test):
        """Check that a decompress attempt via API throws no errors."""
        src_path = scratch_path / (misc_info.FNames.INIT + misc_info.Extensions.CMP)
        dest_path = scratch_path / (misc_info.FNames.MOD + misc_info.Extensions.DEC)

        b_cmp = soi.readbytes(path_fxn(src_path))
        b_dec = soi.decompress(b_cmp)
        soi.writebytes(path_fxn(dest_path), b_dec)

        assert dest_path.is_file()

>       decomp_cmp_test(dest_path)

tests/test_api_good.py:107:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

path = PosixPath('/private/var/folders/y4/n6m6rkls6zs72x8jd2w2tlx40000gn/T/pytest-of-andre/pytest-13/test_api_decompress_no_op_0/objects_mod.txt')

    def func(path):
        """Perform the round-trip compress/decompress comparison test."""
        # The str() calls here are for Python 3.5 compat
>       assert cmp(str(misc_info.res_decomp_path), str(path), shallow=False)
E       AssertionError: assert False
E        +  where False = cmp('tests/resource/objects_attrs.txt', '/private/var/folders/y4/n6m6rkls6zs72x8jd2w2tlx40000gn/T/pytest-of-andre/pytest-13/test_api_decompress_no_op_0/objects_mod.txt', shallow=False)
E        +    where 'tests/resource/objects_attrs.txt' = str(PosixPath('tests/resource/objects_attrs.txt'))
E        +      where PosixPath('tests/resource/objects_attrs.txt') = <conftest.misc_info.<locals>.Info object at 0x102e42250>.res_decomp_path
E        +    and   '/private/var/folders/y4/n6m6rkls6zs72x8jd2w2tlx40000gn/T/pytest-of-andre/pytest-13/test_api_decompress_no_op_0/objects_mod.txt' = str(PosixPath('/private/var/folders/y4/n6m6rkls6zs72x8jd2w2tlx40000gn/T/pytest-of-andre/pytest-13/test_api_decompress_no_op_0/objects_mod.txt'))

conftest.py:269: AssertionError
```

Do I miss a file?

The tests that fail are rather difficult to find in the way pytest is currently configured (I guess this comes from your `tox.ini` file).  I'm not sure why the the failed tests are not printed by the end of the run.

Otherwise, may I suggest you move the file `conftest.py` into the `tests/` directory?  That would help reducing the clutter on the root directory, and copying specific source files on the conda recipe.  So far, I have the following:

* tox.ini
* requirements-dev.txt
* conftest.py
* tests/

--- a/.gitattributes
+++ b/.gitattributes
@@ -1,2 +1,2 @@
 tests/resource/objects_mkdoc_zlib0.inv binary
-
+tests/resource/objects_attrs.txt binary

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -18,6 +18,12 @@ and this project strives to adhere to
   * The 'suggest' CLI mode output now includes dividers for improved
     readability.
 
+#### Tests
+
+  * The README doctests and shell tests have been removed from the default
+    pytest suite. They must be explicitly opted-in with the `--readme` and
+    `--doctest-glob="README.rst"` flags to pytest.
+
 #### Internal
 
   * The `sys.exit()` in the case of no objects falling above the 'suggest' search threshold was refactored into the main `do_suggest()` body, to minimize the surprise of an `exit()` call coming in a subfunction. ([#263](https://github.com/bskinn/sphobjinv/issues/263))

--- a/MANIFEST.in
+++ b/MANIFEST.in
@@ -1,14 +1,15 @@
 include AUTHORS.md CHANGELOG.md CONTRIBUTING.md LICENSE.txt pyproject.toml
 include README.rst requirements-dev.txt requirements-flake8.txt tox.ini
 
-include src/sphobjinv/_vendored/fuzzywuzzy/LICENSE.txt
+graft src/sphobjinv/_vendored/fuzzywuzzy
 
-recursive-include doc/source *
+graft doc/source
 include doc/make.bat doc/Makefile
 
-include tests/*.py
-include tests/resource/objects_attrs*
 include conftest.py
+graft tests
+prune tests/resource
+include tests/resource/objects_attrs* tests/resource/objects_sarge*
 
 global-exclude __pycache__/*
 prune **/*.egg-info

--- a/MANIFEST.in
+++ b/MANIFEST.in
@@ -1,1 +1,14 @@
-include AUTHORS.md CHANGELOG.md CONTRIBUTING.md LICENSE.txt pyproject.toml README.rst
+include AUTHORS.md CHANGELOG.md CONTRIBUTING.md LICENSE.txt pyproject.toml
+include README.rst requirements-dev.txt requirements-flake8.txt tox.ini
+
+include src/sphobjinv/_vendored/fuzzywuzzy/LICENSE.txt
+
+recursive-include doc/source *
+include doc/make.bat doc/Makefile
+
+include tests/*.py
+include tests/resource/objects_attrs*
+include conftest.py
+
+global-exclude __pycache__/*
+prune **/*.egg-info

--- a/azure-coretest.yml
+++ b/azure-coretest.yml
@@ -1,3 +1,6 @@
+# Caching implemented as per
+# https://stackoverflow.com/a/70297887/4376000
+
 parameters:
   pythons: []
   platforms: []
@@ -15,6 +18,7 @@ jobs:
           image: 'macOS-latest'
         ${{ if notIn(platform, 'macOs', 'linux', 'windows') }}:
           image: 'Ubuntu-latest'
+        pip_cache_dir: $(Pipeline.Workspace)/.pip
 
       pool:
         vmImage: $[ variables.image ]
@@ -25,6 +29,14 @@ jobs:
           versionSpec: '${{ python.value.spec }}'
           architecture: 'x64'
         displayName: Use cached ${{ coalesce(python.value.name, python.key) }} for tests.
+
+      - task: Cache@2
+        inputs:
+          key: 'pip | "$(Agent.OS)" | requirements-dev.txt | requirements-flake8.txt'
+          restoreKeys: |
+            pip | "$(Agent.OS)"
+          path: $(pip_cache_dir)
+        displayName: Cache pip
 
       - script: pip install -U --force-reinstall -r requirements-ci.txt
         displayName: Install CI requirements

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -7,6 +7,9 @@ pr:
     - main
     - stable
 
+variables:
+  pip_cache_dir: $(Pipeline.Workspace)/.pip
+
 
 stages:
 
@@ -19,10 +22,19 @@ stages:
 
     pool:
       vmImage: 'Ubuntu-latest'
+
     steps:
     - task: UsePythonVersion@0
       inputs:
         versionSpec: '3.10'
+
+    - task: Cache@2
+      inputs:
+        key: 'pip | "$(Agent.OS)" | requirements-dev.txt | requirements-flake8.txt'
+        restoreKeys: |
+          pip | "$(Agent.OS)"
+        path: $(pip_cache_dir)
+      displayName: Cache pip
 
     - script: pip install -U tox
       displayName: Install tox
@@ -42,10 +54,19 @@ stages:
 
     pool:
       vmImage: 'Ubuntu-latest'
+
     steps:
     - task: UsePythonVersion@0
       inputs:
         versionSpec: '3.10'
+
+    - task: Cache@2
+      inputs:
+        key: 'pip | "$(Agent.OS)" | requirements-dev.txt | requirements-flake8.txt'
+        restoreKeys: |
+          pip | "$(Agent.OS)"
+        path: $(pip_cache_dir)
+      displayName: Cache pip
 
     - script: pip install interrogate
       displayName: Install interrogate
@@ -108,6 +129,14 @@ stages:
       inputs:
         versionSpec: '3.10'
 
+    - task: Cache@2
+      inputs:
+        key: 'pip | "$(Agent.OS)" | requirements-dev.txt | requirements-flake8.txt'
+        restoreKeys: |
+          pip | "$(Agent.OS)"
+        path: $(pip_cache_dir)
+      displayName: Cache pip
+
     - script: pip install . -r requirements-rtd.txt
       displayName: Install project, plus docs requirements
 
@@ -125,6 +154,14 @@ stages:
     - task: UsePythonVersion@0
       inputs:
         versionSpec: '3.10'
+
+    - task: Cache@2
+      inputs:
+        key: 'pip | "$(Agent.OS)" | requirements-dev.txt | requirements-flake8.txt'
+        restoreKeys: |
+          pip | "$(Agent.OS)"
+        path: $(pip_cache_dir)
+      displayName: Cache pip
 
     - script: pip install -r requirements-dev.txt
       displayName: Install full dev requirements
@@ -147,6 +184,14 @@ stages:
     - script: pip install -r requirements-ci.txt
       displayName: Install CI requirements
 
+    - task: Cache@2
+      inputs:
+        key: 'pip | "$(Agent.OS)" | requirements-dev.txt | requirements-flake8.txt'
+        restoreKeys: |
+          pip | "$(Agent.OS)"
+        path: $(pip_cache_dir)
+      displayName: Cache pip
+
     - script: cd doc; make html
       displayName: Build docs
 
@@ -165,6 +210,14 @@ stages:
       inputs:
         versionSpec: '3.10'
 
+    - task: Cache@2
+      inputs:
+        key: 'pip | "$(Agent.OS)" | requirements-dev.txt | requirements-flake8.txt'
+        restoreKeys: |
+          pip | "$(Agent.OS)"
+        path: $(pip_cache_dir)
+      displayName: Cache pip
+
     - script: pip install . -r requirements-rtd.txt
       displayName: Install doc requirements and local project
 
@@ -182,6 +235,14 @@ stages:
     - task: UsePythonVersion@0
       inputs:
         versionSpec: '3.10'
+
+    - task: Cache@2
+      inputs:
+        key: 'pip | "$(Agent.OS)" | requirements-dev.txt | requirements-flake8.txt'
+        restoreKeys: |
+          pip | "$(Agent.OS)"
+        path: $(pip_cache_dir)
+      displayName: Cache pip
 
     - script: pip install -r requirements-dev.txt
       displayName: Install full dev requirements
@@ -207,6 +268,14 @@ stages:
       inputs:
         versionSpec: '3.10'
 
+    - task: Cache@2
+      inputs:
+        key: 'pip | "$(Agent.OS)" | requirements-dev.txt | requirements-flake8.txt'
+        restoreKeys: |
+          pip | "$(Agent.OS)"
+        path: $(pip_cache_dir)
+      displayName: Cache pip
+
     - script: pip install -r requirements-ci.txt
       displayName: Install CI requirements
 
@@ -227,6 +296,14 @@ stages:
     - task: UsePythonVersion@0
       inputs:
         versionSpec: '3.10'
+
+    - task: Cache@2
+      inputs:
+        key: 'pip | "$(Agent.OS)" | requirements-dev.txt | requirements-flake8.txt'
+        restoreKeys: |
+          pip | "$(Agent.OS)"
+        path: $(pip_cache_dir)
+      displayName: Cache pip
 
     - script: pip install tox flake8-noqa -r requirements-flake8.txt
       displayName: Install requirements

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -9,6 +9,79 @@ pr:
 
 
 stages:
+- stage: temp_testable_sdist
+  displayName: Test sdist testability
+
+  jobs:
+  - job: testable_sdist
+    displayName: Ensure sdist is testable
+
+    pool:
+      vmImage: 'Ubuntu-latest'
+
+    steps:
+    - task: UsePythonVersion@0
+      inputs:
+        versionSpec: '3.10'
+
+    # - script: pip install -r requirements-dev.txt
+    #   displayName: Install full dev requirements
+
+    # - script: cd doc; make html
+    #   displayName: Build docs
+
+    - script: python -m pip install build
+      displayName: Install 'build' package
+
+    - script: |
+        python -m build -s
+        ls -lah dist
+      displayName: Build sdist
+
+    - script: |
+        mkdir sandbox
+      displayName: Create sandbox
+
+    - script: |
+        cp dist/*.gz sandbox/
+        cd sandbox
+        tar xf *.gz
+      displayName: Unpack sdist in sandbox
+
+    - script: |
+        cd sandbox
+        python -m venv env
+      displayName: Create venv
+
+    - script: |
+        cd sandbox
+        echo $( find . -maxdepth 1 -type d -regex "./.+[0-9].+" )
+      displayName: Debug cd command arg
+
+    # Only the dir of the unpacked sdist will have a digit in its name
+    - script: |
+        cd sandbox
+        source env/bin/activate
+        cd $( find . -maxdepth 1 -type d -regex "./.+[0-9].+" )
+        python -m pip install -r requirements-dev.txt
+      displayName: Install dev requirements to venv
+
+    - script: |
+        cd sandbox
+        source env/bin/activate
+        cd $( find . -maxdepth 1 -type d -regex "./.+[0-9].+" )
+        cd doc
+        O=-Ean make html
+      displayName: Build docs in sandbox
+
+    - script: |
+        cd sandbox
+        source env/bin/activate
+        cd $( find . -maxdepth 1 -type d -regex "./.+[0-9].+" )
+        pytest --nonloc
+      displayName: Run full test suite in sandbox
+
+
 - stage: code_quality
   displayName: Check code quality
 
@@ -34,6 +107,7 @@ stages:
 
     - script: tox -e flake8
       displayName: Lint the codebase
+
 
   - job: interrogate
     displayName: Run interrogate docstrings check
@@ -128,6 +202,7 @@ stages:
     - script: cd doc; make doctest
       displayName: Run doctests
 
+
   - job: readme
     displayName: Run README doctests/shell tests
 
@@ -147,6 +222,7 @@ stages:
 
     - script: pytest -k readme --readme --doctest-glob="README.rst"
       displayName: Run README doc/shell tests
+
 
   - job: linkcheck
     displayName: Run docs link-check suite
@@ -189,6 +265,7 @@ stages:
     - script: coverage report --include="tests/*" --fail-under=100
       displayName: Check 100% test execution
 
+
   - job: contributing_toc
     displayName: Confirm CONTRIBUTING.md TOC is up to date
 
@@ -208,6 +285,7 @@ stages:
 
     - script: if [[ $( git status | grep modified | wc -l ) -ne 0 ]]; then exit 1; fi
       displayName: Fail if md-toc modified CONTRIBUTING
+
 
   - job: noqa_info
     displayName: Report flake8-noqa results for info (never fails)

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -9,78 +9,6 @@ pr:
 
 
 stages:
-- stage: temp_testable_sdist
-  displayName: Test sdist testability
-
-  jobs:
-  - job: testable_sdist
-    displayName: Ensure sdist is testable
-
-    pool:
-      vmImage: 'Ubuntu-latest'
-
-    steps:
-    - task: UsePythonVersion@0
-      inputs:
-        versionSpec: '3.10'
-
-    # - script: pip install -r requirements-dev.txt
-    #   displayName: Install full dev requirements
-
-    # - script: cd doc; make html
-    #   displayName: Build docs
-
-    - script: python -m pip install build
-      displayName: Install 'build' package
-
-    - script: |
-        python -m build -s
-        ls -lah dist
-      displayName: Build sdist
-
-    - script: |
-        mkdir sandbox
-      displayName: Create sandbox
-
-    - script: |
-        cp dist/*.gz sandbox/
-        cd sandbox
-        tar xf *.gz
-      displayName: Unpack sdist in sandbox
-
-    - script: |
-        cd sandbox
-        python -m venv env
-      displayName: Create venv
-
-    - script: |
-        cd sandbox
-        echo $( find . -maxdepth 1 -type d -regex "./.+[0-9].+" )
-      displayName: Debug cd command arg
-
-    # Only the dir of the unpacked sdist will have a digit in its name
-    - script: |
-        cd sandbox
-        source env/bin/activate
-        cd $( find . -maxdepth 1 -type d -regex "./.+[0-9].+" )
-        python -m pip install -r requirements-dev.txt
-      displayName: Install dev requirements to venv
-
-    - script: |
-        cd sandbox
-        source env/bin/activate
-        cd $( find . -maxdepth 1 -type d -regex "./.+[0-9].+" )
-        cd doc
-        O=-Ean make html
-      displayName: Build docs in sandbox
-
-    - script: |
-        cd sandbox
-        source env/bin/activate
-        cd $( find . -maxdepth 1 -type d -regex "./.+[0-9].+" )
-        pytest --nonloc
-      displayName: Run full test suite in sandbox
-
 
 - stage: code_quality
   displayName: Check code quality
@@ -161,6 +89,8 @@ stages:
         py311:
           spec: '3.11'
       platforms: [windows, macOs]
+
+  - template: azure-sdisttest.yml
 
 
 - stage: aux_tests

--- a/azure-sdisttest.yml
+++ b/azure-sdisttest.yml
@@ -2,6 +2,9 @@ jobs:
 - job: testable_sdist
   displayName: Ensure sdist is testable
 
+  variables:
+    pip_cache_dir: $(Pipeline.Workspace)/.pip
+
   pool:
     vmImage: 'Ubuntu-latest'
 
@@ -9,6 +12,14 @@ jobs:
   - task: UsePythonVersion@0
     inputs:
       versionSpec: '3.10'
+
+  - task: Cache@2
+    inputs:
+      key: 'pip | "$(Agent.OS)" | requirements-dev.txt | requirements-flake8.txt'
+      restoreKeys: |
+        pip | "$(Agent.OS)"
+      path: $(pip_cache_dir)
+    displayName: Cache pip
 
   - script: python -m pip install build
     displayName: Install 'build' package

--- a/azure-sdisttest.yml
+++ b/azure-sdisttest.yml
@@ -1,0 +1,62 @@
+jobs:
+- job: testable_sdist
+  displayName: Ensure sdist is testable
+
+  pool:
+    vmImage: 'Ubuntu-latest'
+
+  steps:
+  - task: UsePythonVersion@0
+    inputs:
+      versionSpec: '3.10'
+
+  - script: python -m pip install build
+    displayName: Install 'build' package
+
+  - script: |
+      python -m build -s
+      ls -lah dist
+    displayName: Build sdist
+
+  - script: |
+      mkdir sandbox
+    displayName: Create sandbox
+
+  - script: |
+      cp dist/*.gz sandbox/
+      cd sandbox
+      tar xvf *.gz
+    displayName: Unpack sdist in sandbox
+
+  - script: |
+      cd sandbox
+      python -m venv env
+    displayName: Create venv
+
+  # Only the dir of the unpacked sdist will have a digit in its name
+  - script: |
+      cd sandbox
+      echo $( find . -maxdepth 1 -type d -regex "./.+[0-9].+" )
+    displayName: Check unpack dir name
+
+  - script: |
+      cd sandbox
+      source env/bin/activate
+      cd $( find . -maxdepth 1 -type d -regex "./.+[0-9].+" )
+      python -m pip install -r requirements-dev.txt
+    displayName: Install dev req'ts to venv
+
+  - script: |
+      cd sandbox
+      source env/bin/activate
+      cd $( find . -maxdepth 1 -type d -regex "./.+[0-9].+" )
+      cd doc
+      O=-Ean make html
+    displayName: Build docs in sandbox
+
+  - script: |
+      cd sandbox
+      source env/bin/activate
+      cd $( find . -maxdepth 1 -type d -regex "./.+[0-9].+" )
+      pytest --nonloc
+    displayName: Run test suite in sandbox

--- a/conftest.py
+++ b/conftest.py
@@ -36,7 +36,6 @@ import re
 import shutil
 import sys
 from enum import Enum
-from filecmp import cmp
 from functools import partial
 from io import BytesIO
 from pathlib import Path
@@ -139,7 +138,7 @@ def misc_info(res_path):
 
 
 @pytest.fixture()
-def scratch_path(tmp_path, res_path, misc_info):
+def scratch_path(tmp_path, res_path, misc_info, is_win, unix2dos):
     """Provision pre-populated scratch directory, returned as Path."""
     res_base = misc_info.FNames.RES.value
     scr_base = misc_info.FNames.INIT.value
@@ -150,6 +149,13 @@ def scratch_path(tmp_path, res_path, misc_info):
             str(res_path / f"{res_base}{ext}"),
             str(tmp_path / f"{scr_base}{ext}"),
         )
+
+    # With the conversion of resources/objects_attrs.txt to Unix EOLs in order to
+    # provide for a Unix-testable sdist, on Windows systems this resource needs
+    # to be converted to DOS EOLs for consistency.
+    if is_win:
+        win_path = tmp_path / f"{scr_base}{misc_info.Extensions.DEC}"
+        win_path.write_bytes(unix2dos(win_path.read_bytes()))
 
     yield tmp_path
 
@@ -260,13 +266,22 @@ def run_cmdline_test(monkeypatch):
 
 
 @pytest.fixture(scope="session")
-def decomp_cmp_test(misc_info):
+def decomp_cmp_test(misc_info, is_win, unix2dos):
     """Return function to confirm a decompressed file is identical to resource."""
 
     def func(path):
         """Perform the round-trip compress/decompress comparison test."""
         # The str() calls here are for Python 3.5 compat
-        assert cmp(str(misc_info.res_decomp_path), str(path), shallow=False)
+        res_bytes = Path(misc_info.res_decomp_path).read_bytes()
+        tgt_bytes = Path(path).read_bytes()  # .replace(b"\r\n", b"\n")
+
+        if is_win:
+            # Have to explicitly convert these newlines, now that the
+            # tests/resource/objects_attrs.txt file is marked 'binary' in
+            # .gitattributes
+            res_bytes = unix2dos(res_bytes)
+
+        assert res_bytes == tgt_bytes
 
     return func
 

--- a/conftest.py
+++ b/conftest.py
@@ -154,7 +154,7 @@ def scratch_path(tmp_path, res_path, misc_info, is_win, unix2dos):
     # provide for a Unix-testable sdist, on Windows systems this resource needs
     # to be converted to DOS EOLs for consistency.
     if is_win:
-        win_path = tmp_path / f"{scr_base}{misc_info.Extensions.DEC}"
+        win_path = tmp_path / f"{scr_base}{misc_info.Extensions.DEC.value}"
         win_path.write_bytes(unix2dos(win_path.read_bytes()))
 
     yield tmp_path

--- a/doc/source/index.rst
+++ b/doc/source/index.rst
@@ -57,7 +57,7 @@ and package managers, including:
 
   * Gentoo: ``dev-python/sphobjinv`` (`info <https://gitweb.gentoo.org/repo/gentoo.git/tree/dev-python/sphobjinv>`__)
 
-  * Guix: ``python-sphobjinv`` (`info <https://guix.gnu.org/en/packages/python-sphobjinv-2.0.1/>`__)
+  * Guix: ``python-sphobjinv`` (`info <https://packages.guix.gnu.org/packages/python-sphobjinv/2.0.1/>`__)
 
   * Manjaro: ``python-sphobjinv``
 

--- a/src/sphobjinv/version.py
+++ b/src/sphobjinv/version.py
@@ -29,4 +29,4 @@ Sphinx |objects.inv| files.
 
 """
 
-__version__ = "2.3.1.dev1"
+__version__ = "2.3.1.dev2"

--- a/src/sphobjinv/version.py
+++ b/src/sphobjinv/version.py
@@ -29,4 +29,4 @@ Sphinx |objects.inv| files.
 
 """
 
-__version__ = "2.3.1.dev0"
+__version__ = "2.3.1.dev1"

--- a/src/sphobjinv/version.py
+++ b/src/sphobjinv/version.py
@@ -29,4 +29,4 @@ Sphinx |objects.inv| files.
 
 """
 
-__version__ = "2.3.1.dev2"
+__version__ = "2.3.1.dev3"

--- a/tests/test_fixture.py
+++ b/tests/test_fixture.py
@@ -63,11 +63,13 @@ def test_cli_invoke(run_cmdline_test):
     run_cmdline_test([])
 
 
-def test_decomp_comp_fixture(misc_info, decomp_cmp_test):
+def test_decomp_comp_fixture(misc_info, decomp_cmp_test, scratch_path):
     """Test decomp_cmp_test works in 'identity' case.
 
     Basically is telling filecmp.cmp to compare a reference inventory file
     with itself.
 
     """
-    decomp_cmp_test(misc_info.res_decomp_path)
+    decomp_cmp_test(
+        scratch_path / f"{misc_info.FNames.INIT.value}{misc_info.Extensions.DEC}"
+    )

--- a/tests/test_fixture.py
+++ b/tests/test_fixture.py
@@ -71,5 +71,5 @@ def test_decomp_comp_fixture(misc_info, decomp_cmp_test, scratch_path):
 
     """
     decomp_cmp_test(
-        scratch_path / f"{misc_info.FNames.INIT.value}{misc_info.Extensions.DEC}"
+        scratch_path / f"{misc_info.FNames.INIT.value}{misc_info.Extensions.DEC.value}"
     )

